modules/031-linstor/templates/piraeus-operator/deployment.yaml

```diff
@@ -0,0 +1,86 @@
+# Source: https://github.com/piraeusdatastore/piraeus-operator/blob/master/charts/piraeus/templates/operator-deployment.yaml
+{{- if (.Values.global.enabledModules | has "vertical-pod-autoscaler-crd") }}
+---
+apiVersion: autoscaling.k8s.io/v1
+kind: VerticalPodAutoscaler
+metadata:
+  name: piraeus-operator
+  namespace: d8-{{ .Chart.Name }}
+  {{- include "helm_lib_module_labels" (list . (dict "app" "piraeus-operator" "workload-resource-policy.deckhouse.io" "every-node")) | nindent 2 }}
+spec:
+  targetRef:
+    apiVersion: "apps/v1"
+    kind: Deployment
+    name: piraeus-operator
+  updatePolicy:
+    updateMode: "Auto"
+{{- end }}
+---
+apiVersion: policy/v1beta1
+kind: PodDisruptionBudget
+metadata:
+  annotations:
+    helm.sh/hook: post-upgrade, post-install
+    helm.sh/hook-delete-policy: before-hook-creation
+  name: piraeus-operator
+  namespace: d8-{{ .Chart.Name }}
+  {{- include "helm_lib_module_labels" (list . (dict "app" "piraeus-operator" )) | nindent 2 }}
+spec:
+  minAvailable: {{ include "helm_lib_is_ha_to_value" (list . 1 0) }}
+  selector:
+    matchLabels:
+      app: piraeus-operator
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: piraeus-operator
+  namespace: d8-{{ .Chart.Name }}
+  {{- include "helm_lib_module_labels" (list . (dict "app" "piraeus-operator" )) | nindent 2 }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: piraeus-operator
+  template:
+    metadata:
+      labels:
+        app: piraeus-operator
+    spec:
+      {{- include "helm_lib_priority_class" (tuple . "system-cluster-critical") | nindent 6 }}
```

Comment:
What will happen if the operator is unavailable? Are you sure that the "system-cluster-critical" priority class should be used?

Reply:
Piraeus-operator is used not only to generate Kubernetes-resources but also for registering new nodes in linstor and handle their eviction.

Reply:
It still does not seem to be system critical. We are trying to prioritize pods eviction, and if we can not add new nodes yet, prod is still working - it is not so critical. Even Deckhouse itself has the cluster-medium priority class.

Reply:
PiorityClass for operator changed to `cluster-medium`, what do you think about doing the same for `ha-controller`, this entity used to deschedule pod from the failed nodes in the cluster, which enables optional storage-based fencing for the statefulset pods?